pkg/service/service.go

```diff
@@ -2,18 +2,105 @@ package service
 
 import (
 	"context"
-	"errors"
+	"time"
 
+	"github.com/Merteg/pl-health-service/config"
+	"github.com/Merteg/pl-health-service/pkg/model"
 	"github.com/Merteg/pl-health-service/proto"
+	"github.com/rs/zerolog/log"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gopkg.in/mgo.v2/bson"
 )
 
 type Health struct {
 	proto.UnimplementedHealthServiceServer
 }
 
-func (s *Health) Push(context.Context, *proto.PushRequest) (*proto.PushResponse, error) {
-	return nil, errors.New("not implemented")
+var mongoconfig = config.GetConfig().Mongo
+
+func (s *Health) Push(c context.Context, req *proto.PushRequest) (*proto.PushResponse, error) {
+	client, ctx := MongodbClient()
+	healthcollection := client.Database(mongoconfig["dbname"]).Collection(mongoconfig["healthcollname"])
+	var health []*proto.Health = req.GetHealth()
+
+	for _, reqhealth := range health {
+		id := reqhealth.TargetID
+
+		var resphealth model.Health
+
+		error := healthcollection.FindOne(context.TODO(), bson.M{"targetid": id}).Decode(&resphealth)
+		if error != nil {
+			log.Fatal().Err(error)
+		}
+
+		if resphealth.TargetID == "" {
+			targetcollection := client.Database(mongoconfig["dbname"]).Collection(mongoconfig["targetscollname"])
+			var target model.Target
+
+			error := targetcollection.FindOne(context.TODO(), bson.M{"_id": id}).Decode(&target)
+			if error != nil {
+				log.Fatal().Err(error)
+			}
+			resphealth.FromProto(reqhealth)
+
+			_, err := healthcollection.InsertOne(ctx, resphealth)
+			if err != nil {
+				log.Fatal().Err(err)
+			}
+		} else {
+			status.Error(codes.AlreadyExists, "This TargetID already exist")
```

Comment:
Wrong logic.
You need to check if target exists in Target's collection and return an error if it is not
If it is exist we just save the health metrics in health collection